halo2_proofs/src/dev/cost_model.rs

```diff
@@ -353,16 +353,25 @@ pub fn from_circuit_to_cost_model_options<F: Ord + Field + FromUniformBytes<64>,
         let mut expanded_rows_count = 0;
         let mut compressed_rows_count = 0;
         for region in prover.regions {
-            if region.name.contains("table") { // Account for lookup tables separately
-                let (start, end) = region.rows.expect("region.rows should be known by now");
-                min_k = std::cmp::max(min_k, end - start);
-            } else {
-                let (start, end) = region.rows.expect("region.rows should be known by now");
-                expanded_rows_count = expanded_rows_count + ((end - start) + 1);
-                compressed_rows_count = std::cmp::max(compressed_rows_count, end + 1);
-            }
+            // If `region.rows == None`, then that region has no rows (e.g. just copy constraints)
+            if let Some((start, end)) = region.rows {
+                // A region is a _table region_ if all of its columns are `Fixed`
+                // columns (see that [`plonk::circuit::TableColumn` is a wrapper
+                // around `Column<Fixed>`]). All of a table region's rows are
+                // counted towards `table_rows_count.`
```

Comment:
Is this characterization always correct?

Reply:
This is a good question. What I used before was in the naming, but that is also error prone. This would certainly not work with dynamic lookups, but with static lookups, my understanding is that their region consists only of the table (not the columns that make queries to it).